### DISCUSSION TITLE
Provide a proxy for each item  like ItemProvider

### DISCUSSION THREAD

**Discussion by amjadbouhouch** [pmndrs/valtio#352] [Q&A] [answered]
First of all thank you for this great package

I'm just getting started today with valtio and I like the idea and the approach 

I'm currently using context in react and when I have a list with **complex item manupilation I wrapped it within a context** .  
I think valtio have different approach to achieve this ? Or should I use the global proxy that manipulate all the list.

Thanks.

**Comment by dai-shi** [comment, marked as answer]
In valtio, all nested objects are proxies.
https://github.com/pmndrs/valtio/wiki/How-to-split-and-compose-states
Let me know if your question is something else.

**Reply by amjadbouhouch** [reply]
Thank you @dai-shi that's what I looking for.
